stellar-multitenant/internal/validators/tenant_validator.go

```diff
@@ -92,6 +84,18 @@ func (tv *TenantValidator) ValidateCreateTenantRequest(reqBody *TenantRequest) *
 	return reqBody
 }
 
+func (tv *TenantValidator) validateDistributionAccountType(distributionAccountType string) {
+	tv.Check(distributionAccountType != "", "distribution_account_type", fmt.Sprintf("distribution_account_type is required. valid values are: %v", schema.DistributionAccountTypes()))
+
+	if distributionAccountType != "" {
+		switch schema.AccountType(distributionAccountType) {
+		case schema.DistributionAccountStellarEnv, schema.DistributionAccountStellarDBVault, schema.DistributionAccountCircleDBVault:
+		default:
+			tv.Check(false, "distribution_account_type", fmt.Sprintf("invalid distribution_account_type. valid values are: %v", schema.DistributionAccountTypes()))
+		}
+	}
+}
+
```

Comment:
💡 you could simplify this a bit:

```suggestion
func (tv *TenantValidator) validateDistributionAccountType(distributionAccountType string) {
	if slices.Contains(schema.DistributionAccountTypes(), schema.AccountType(distributionAccountType)) {
		return
	}
	tv.Check(false, "distribution_account_type", fmt.Sprintf("invalid distribution_account_type, the valid values are: %v", schema.DistributionAccountTypes()))
}
```

Reply:
I still want to differentiate between when the field is missing vs. invalid, but I'll take your suggestion using the slice directly rather than enumerating the cases through all the values using switch-case.